OpenEdXMobile/src/main/java/org/edx/mobile/util/WebViewUtil.java

```diff
@@ -31,4 +55,87 @@ public static StringBuilder getIntialWebviewBuffer(Context context, Logger logge
         buff.append("</head>");
         return buff;
     }
+
+    /**
+     * Clears content of the WebView and loads {@link #EMPTY_HTML} in it.
+     */
+    public static void clearWebviewHtml(@Nullable WebView webView) {
+        if (webView != null) {
+            webView.loadData(EMPTY_HTML, "text/html", "UTF-8");
+        }
+    }
+
+    /**
+     * Simply loads a url within a WebView for Marshmallow & above and differently in case of
+     * Lollipop & below.<br/>
+     * WebViews prior to Marshmallow (API Level 23) don't provide a way to get the HTTP status
+     * codes if the url being loaded fails.<br/>
+     * This utility function solves this by making a server call using the url provided to query if
+     * an error is being return and show error or move on to load the url in WebView, if the server
+     * is responding correctly.
+     * <p>
+     * Inspiration for this solution has been taken from this link:
+     * https://stackoverflow.com/questions/11889020/get-http-status-code-in-android-webview/21609608#21609608
+     *
+     * @param context              Current context.
+     * @param webView              The WebView to load the URL into.
+     * @param url                  The URL to load.
+     * @param viewInterface        WebView's callbacks interface.
+     * @param errorNotification    The notification setup for showing/hiding errors.
+     * @param okHttpClientProvider The utility to make server calls.
+     */
+    public static void loadUrlBasedOnOsVersion(@NonNull final Context context,
+                                               @NonNull final WebView webView,
+                                               @NonNull final String url,
+                                               @NonNull final WebViewStatusListener viewInterface,
+                                               @NonNull final FullScreenErrorNotification errorNotification,
+                                               @NonNull OkHttpClientProvider okHttpClientProvider) {
+        if (!NetworkUtil.isConnected(context)) {
+            errorNotification.showError(context, new IOException());
+        } else {
+            errorNotification.hideError();
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+                viewInterface.showLoadingProgress();
+                okHttpClientProvider.get().newCall(new Request.Builder()
+                        .url(url)
+                        .get()
+                        .build())
+                        .enqueue(new Callback() {
+                            @Override
+                            public void onFailure(Call call, final IOException e) {
+                                HANDLER.post((new Runnable() {
```

Comment:
You can simply use a following callback rather than creating new static handler object.
                                
```
                                webView.post(
                                        new Runnable() {
                                            @Override
                                            public void run() {
                                                
                                            }
                                        }
                                );

```